docs/sources/operations/shuffle-sharding.md

```diff
@@ -0,0 +1,75 @@
+
+A misbehaving tenant will affect only its shard instances. Due to the low overlap of instances between different tenants, it’s statistically quite likely that any other tenant will run on different instances or only a subset of instances will match the affected ones.
+Shuffle sharding requires no more resources than the default sharding strategy but instances may be less evenly balanced from time to time.
+
+### Low overlapping instances probability
+For example, given a Loki cluster running 50 queriers and assigning each tenant 4 out of 50 queriers, shuffling instances between each tenant, we get 230K possible combinations.
+
+Randomly picking two different tenants we have the:
+
+- 71% chance that they will not share any instance
+- 26% chance that they will share only 1 instance
+- 2.7% chance that they will share 2 instances
+- 0.08% chance that they will share 3 instances
+- Only a 0.0004% chance that their instances will fully overlap
+
+![overlapping instances probability](./shuffle-sharding-probability.png)
+
+## Why we need it on Loki?
+We use shuffle sharding only on **Query** path in Loki.
+
+In the default sharding strategy employed by Loki query path the workload uses the entire pool of instances running a given service (eg. queriers). For example, on the query path each tenant’s queries are sharded across all queriers, regardless how many different tenants are in the cell.
+
+The default strategy allows to have a fair balance on the resources consumed by each instance (ie. CPU and memory) and to maximise these resources across the cluster.
+
+However, in a multi-tenant cluster this approach also introduces some downsides:
+- An outage of node(s) affects all tenants
+- A misbehaving tenant (eg. causing out of memory) could affect all other tenants
+
+The goal of shuffle sharding is to provide an alternative sharding strategy to reduce the blast radius of an outage and better isolate tenants.
+
+By default all Loki queriers can execute received queries for any given tenant.
+
+When shuffle sharding is enabled by setting `-frontend.max-queriers-per-tenant` to a value higher than 0 and lower than the number of available queriers, only specified number of queriers will execute queries for single tenant.
+
+NOTE: this option is only available when using the query-frontend (with or without scheduler).
+
+The maximum number of queriers can be overridden on a per-tenant basis in the limits overrides configuration (`max_queriers_per_tenant`).
+
+### The impact of “query of death”
+In the event a tenant is repeatedly sending a “query of death” which leads the querier to crash or getting killed because of out-of-memory, the crashed querier will get disconnected from the query-frontend and a new querier will be immediately assigned to the tenant’s shard. This practically invalidates the assumption that shuffle-sharding can be used to contain the blast radius in case of a query of death.
```

Comment:
```suggestion
In the event a tenant is repeatedly sending a “query of death” which leads the querier to crash or get killed because of out-of-memory, the crashed querier will get disconnected from the query-frontend and a new querier will be immediately assigned to the tenant’s shard. This practically invalidates the assumption that shuffle-sharding can be used to contain the blast radius in case of a query of death.
```